### PR TITLE
[zh-cn] Update 'chrome incompatibilities'

### PR DESCRIPTION
**Pull request by jasonren0403**
### Description

Finish half translated files

### CODE REVIEW

File: files/zh-cn/mozilla/add-ons/webextensions/chrome_incompatibilities/index.md [hunk 1]
```diff
@@ -1,62 +1,84 @@
...
 
-使用 WebExtension API 构建的扩展旨在与 Chrome 和 Opera 扩展兼容：尽可能为这些浏览器编写的扩展应该在 Firefox 上运行，并且只需进行极少的更改。
+WebExtension API 的目标是提供对所有主要浏览器的兼容性，因此扩展程序只需最小的改动即可在任何浏览器上运行。
 
-但是，Firefox 目前仅支持 Chrome 和 Opera 支持的有限功能和 API。我们正在努力增加更多的支持，但许多功能尚未得到支持，我们可能永远不会支持某些功能。
+不过，Chrome 浏览器（以及基于 Chromium 的浏览器）、Firefox 浏览器和 Safari 浏览器之间存在很大差异。特别是：
 
-## JavaScript APIs
+- 不同浏览器对 WebExtension API 的支持各不相同。详情请参阅[浏览器对 JavaScript API 的支持](/zh-CN/docs/Mozilla/Add-ons/WebExtensions/Browser_support_for_JavaScript_APIs)。
+- 不同浏览器对 `manifest.json` 键的支持有所不同。请参阅 [`manifest.json``](/zh-CN/docs/Mozilla/Add-ons/WebExtensions/manifest.json#browser_compatibility) 页面上的[“浏览器兼容性”部分](/zh-CN/docs/Mozilla/Add-ons/WebExtensions/manifest.json)了解更多详情。
```
Comment (jasonren0403) [inline files/zh-cn/mozilla/add-ons/webextensions/chrome_incompatibilities/index.md:15]:
```suggestion
- 不同浏览器对 `manifest.json` 键的支持有所不同。请参阅 [`manifest.json``](/zh-CN/docs/Mozilla/Add-ons/WebExtensions/manifest.json#浏览器兼容性) 页面上的[“浏览器兼容性”部分](/zh-CN/docs/Mozilla/Add-ons/WebExtensions/manifest.json)了解更多详情。
```

File: files/zh-cn/mozilla/add-ons/webextensions/chrome_incompatibilities/index.md [hunk 2]
```diff
@@ -66,80 +88,171 @@ setCookie.then(logCookie, logError);
...
+
+- **在 Chrome 中**：应用程序清单预计会放在不同的位置。请参阅 Chrome 文档中的[本地消息主机位置](https://developer.chrome.com/docs/apps/nativeMessaging/#native-messaging-host-location)。
+
+### 应用持久化
+
+- **在 Firefox 中**：本机消息传输连接关闭时，如果子进程没有脱离，Firefox 会将其杀死。在 Windows 系统中，浏览器会将本机应用程序的进程放入[作业对象](<https://msdn.microsoft.com/library/windows/desktop/ms684161(v=vs.85).aspx>)并杀死该作业。假设本地应用程序启动了其他进程，并希望这些进程在本地应用程序被杀死后继续打开。在这种情况下，本地应用程序必须使用 `CreateProcess` 而不是 `ShellExecute` 来启动其他进程，并使用 [`CREATE_BREAKAWAY_FROM_JOB`](<https://msdn.microsoft.com/library/windows/desktop/ms684863(v=vs.85).aspx>) 标志。
+
+## 数据克隆算法
+
+某些扩展 API 允许扩展从扩展的一部分向另一部分发送数据，例如 {{WebExtAPIRef("runtime.sendMessage()")}}、{{WebExtAPIRef("tabs. sendMessage()")}}、{{WebExtAPIRef("runtime.onMessage")}}、{{WebExtAPIRef("runtime.port")}} 的 `postMessage()` 方法和 {{WebExtAPIRef("tabs.executeScript()")}}。
```
Comment (jasonren0403) [inline files/zh-cn/mozilla/add-ons/webextensions/chrome_incompatibilities/index.md:251]:
```suggestion
某些扩展 API 允许扩展从扩展的一部分向另一部分发送数据，例如 {{WebExtAPIRef("runtime.sendMessage()")}}、{{WebExtAPIRef("tabs.sendMessage()")}}、{{WebExtAPIRef("runtime.onMessage")}}、{{WebExtAPIRef("runtime.port")}} 的 `postMessage()` 方法和 {{WebExtAPIRef("tabs.executeScript()")}}。
```

File: files/zh-cn/mozilla/add-ons/webextensions/chrome_incompatibilities/index.md [hunk 3]
```diff
@@ -66,80 +88,171 @@ setCookie.then(logCookie, logError);
...
+    - **从 Firefox 57 开始**：Firefox 对需要拦截 {{WebExtAPIRef("webRequest.onAuthRequired")}} 代理授权的扩展例外处理。请参阅 {{WebExtAPIRef("webRequest.onAuthRequired")}} 的文档。
 
-- Firefox 中 `onFocusChanged` 对于指定的焦点变化将触发多次。
+  - 如果扩展希望将公共（如 HTTPS）URL 重定向到[扩展页面](/zh-CN/docs/Mozilla/Add-ons/WebExtensions/user_interface/Extension_pages)，扩展的 `manifest.json` 文件必须包含 [`web_accessible_resources`](/zh-CN/docs/Mozilla/Add-ons/WebExtensions/manifest.json/web_accessible_resources) 键，其中包含扩展页面的 URL。
+
+    - > **备注：** 任何网站都可以链接或重定向到该 URL，扩展应将任何输入（例如 POST 数据）视为来自不可信任的来源，就像普通网页一样。
+
+  - 某些 `browser.webRequest.*` API 允许异步返回解析为 `webRequest.BlockingResponse` 的 Promise。
+
+- **在 Chrome 中**：只有 `webRequest.onAuthRequired` 通过提供 `'asyncBlocking'`，通过回调而非 Promise 支持异步 `webRequest.BlockingResponse`。
```
Comment (yin1999) [inline files/zh-cn/mozilla/add-ons/webextensions/chrome_incompatibilities/index.md:152]:
```suggestion
- **在 Chrome 中**：只有在提供 `'asyncBlocking'` 选项的情况下，`webRequest.onAuthRequired` 才（通过回调而非 Promise）支持异步的 `webRequest.BlockingResponse`。
```

File: files/zh-cn/mozilla/add-ons/webextensions/chrome_incompatibilities/index.md [hunk 4]
```diff
@@ -66,80 +88,171 @@ setCookie.then(logCookie, logError);
...
+
+#### Windows API
+
+- **在 Firefox 中**：{{WebExtAPIRef("windows")}} API 的 `onFocusChanged` 对于指定的焦点变化将触发多次。
+
+### 不支持的 API
+
+#### DeclarativeContent API
+
+- **在 Firefox 中**：Chrome 浏览器的 [declarativeContent](https://developer.chrome.com/docs/extensions/reference/declarativeContent/) API [未实现](https://bugzil.la/1435864)。此外，Firefox [将不支持](https://bugzil.la/1323433#c16) `declarativeContent.RequestContentScript` API（该 API 很少使用，在 Chrome 浏览器的稳定版本中也不可用）。
```
Comment (yin1999) [inline files/zh-cn/mozilla/add-ons/webextensions/chrome_incompatibilities/index.md:162]:
```suggestion
- **在 Firefox 中**：Chrome 浏览器的 [declarativeContent](https://developer.chrome.com/docs/extensions/reference/declarativeContent/) API [未被实现](https://bugzil.la/1435864)。此外，Firefox [将不支持](https://bugzil.la/1323433#c16) `declarativeContent.RequestContentScript` API（该 API 很少使用，在 Chrome 浏览器的稳定版本中也不可用）。
```

File: files/zh-cn/mozilla/add-ons/webextensions/chrome_incompatibilities/index.md [hunk 5]
```diff
@@ -66,80 +88,171 @@ setCookie.then(logCookie, logError);
...
+
+### 应用持久化
+
+- **在 Firefox 中**：本机消息传输连接关闭时，如果子进程没有脱离，Firefox 会将其杀死。在 Windows 系统中，浏览器会将本机应用程序的进程放入[作业对象](<https://msdn.microsoft.com/library/windows/desktop/ms684161(v=vs.85).aspx>)并杀死该作业。假设本地应用程序启动了其他进程，并希望这些进程在本地应用程序被杀死后继续打开。在这种情况下，本地应用程序必须使用 `CreateProcess` 而不是 `ShellExecute` 来启动其他进程，并使用 [`CREATE_BREAKAWAY_FROM_JOB`](<https://msdn.microsoft.com/library/windows/desktop/ms684863(v=vs.85).aspx>) 标志。
+
+## 数据克隆算法
+
+某些扩展 API 允许扩展从扩展的一部分向另一部分发送数据，例如 {{WebExtAPIRef("runtime.sendMessage()")}}、{{WebExtAPIRef("tabs.sendMessage()")}}、{{WebExtAPIRef("runtime.onMessage")}}、{{WebExtAPIRef("runtime.port")}} 的 `postMessage()` 方法和 {{WebExtAPIRef("tabs.executeScript()")}}。
+
+- **在 Firefox 中**：使用 [结构化克隆算法](/zh-CN/docs/Web/API/Web_Workers_API/Structured_clone_algorithm)。
```
Comment (yin1999) [inline files/zh-cn/mozilla/add-ons/webextensions/chrome_incompatibilities/index.md:253]:
```suggestion
- **在 Firefox 中**：使用[结构化克隆算法](/zh-CN/docs/Web/API/Web_Workers_API/Structured_clone_algorithm)。
```

File: files/zh-cn/mozilla/add-ons/webextensions/manifest.json/browser_specific_settings/index.md [hunk 6]
```diff
@@ -17,19 +17,19 @@ slug: Mozilla/Add-ons/WebExtensions/manifest.json/browser_specific_settings
...
       <td>
         通常不做强制性要求（详细参见<a
-          href="/zh-CN/Add-ons/WebExtensions/WebExtensions_and_the_Add-on_ID#When_do_you_need_an_Add-on_ID"
-          >When do you need an Add-on ID</a
-        ><a
-          href="/zh-CN/Add-ons/WebExtensions/manifest.json/applications#When_do_I_need_the_applications_key"
-          >?</a
-        >)。在 Firefox 48 版本之前为强制要求。
+          href="https://extensionworkshop.com/documentation/develop/extensions-and-the-add-on-id/#when-do-you-need-an-add-on-id"
+          >什么时候需要加载项 ID？</a
```
Comment (yin1999) [inline files/zh-cn/mozilla/add-ons/webextensions/manifest.json/browser_specific_settings/index.md:21]:
```suggestion
          >什么时候需要附加组件 ID？</a
```

File: files/zh-cn/mozilla/add-ons/webextensions/manifest.json/browser_specific_settings/index.md [hunk 7]
```diff
@@ -41,30 +41,36 @@ slug: Mozilla/Add-ons/WebExtensions/manifest.json/browser_specific_settings
...
-`applications` key 包含特定于主机应用程序的键。
+## 描述
 
-目前只包含了一个 key：`gecko`，其中包含了 4 个 attributes:
+`browser_specific_settings` 键包含特定于主机应用程序的键。
 
-- `id`即[extension ID](/zh-CN/Add-ons/Install_Manifests#id)。从 Firefox 48 起为可选项，在此之前为强制必须项。参看[WebExtensions and the Add-on ID](/zh-CN/docs/Mozilla/Add-ons/WebExtensions/WebExtensions_and_the_Add-on_ID)来确认何时需要定义附加组件 ID。
+### Firefox（Gecko）属性
+
+- `id` 即扩展 ID。从 Firefox 48 起为可选项，在此之前为强制必须项。参看 [WebExtensions 和加载项 ID](https://extensionworkshop.com/documentation/develop/extensions-and-the-add-on-id/) 来确认何时需要定义附加组件 ID。
```
Comment (yin1999) [inline files/zh-cn/mozilla/add-ons/webextensions/manifest.json/browser_specific_settings/index.md:50]:
```suggestion
- `id` 即扩展 ID。从 Firefox 48 起为可选项，在此之前为强制必须项。参看 [WebExtensions 和附加组件 ID](https://extensionworkshop.com/documentation/develop/extensions-and-the-add-on-id/) 来确认何时需要定义附加组件 ID。
```

File: files/zh-cn/mozilla/add-ons/webextensions/manifest.json/browser_specific_settings/index.md [hunk 8]
```diff
@@ -41,30 +41,36 @@ slug: Mozilla/Add-ons/WebExtensions/manifest.json/browser_specific_settings
...
 - `strict_min_version`: Gecko 所能支持的最小版本号。不允许使用"\*"来定义版本号。默认值为 "42a1"。
 - `strict_max_version`: Gecko 所能支持的最大版本号。如果安装或运行附加组件的 Firefox 版本号高于这个最大版本号，附加组件将不能运行或不允许被安装。默认值为"\*"，意思为不对最大版本号做检查。
-- `update_url`为链接到[add-on update manifest](/zh-CN/Add-ons/Updates)的链接。注意链接必须以"https"开头。这是为了使你自己就能够管理附加组件的更新 (如不通过 AMO)。
-
-## Chrome 不兼容性
-
-你不能在 Chrome 附加组件中使用这个 key。
+- `update_url` 为链接到 [add-on update manifest](https://extensionworkshop.com/documentation/manage/updating-your-extension/) 的链接。注意链接必须以"https"开头。这是为了使你自己就能够管理附加组件的更新（如不通过 AMO）。
```
Comment (yin1999) [inline files/zh-cn/mozilla/add-ons/webextensions/manifest.json/browser_specific_settings/index.md:53]:
```suggestion
- `update_url` 为链接到[扩展更新清单](https://extensionworkshop.com/documentation/manage/updating-your-extension/)的链接。注意链接必须以“https”开头。这是为了使你自己就能够管理附加组件的更新（如不通过 AMO）。
```

**Review by yin1999 — COMMENTED** [review]
Some suggestions

**Review by yin1999 — APPROVED** [review]
Thanks @jasonren0403. Feel free to merge it after the upstream changes are settled

**Pull request merged**